### PR TITLE
fix(electric): Implement support for AuthenticationMD5Password upstream auth method

### PR DESCRIPTION
**Pull request by alco**
Fixes VAX-1553.

@magnetised I would appreciate any guidance on how to best test support for specific upstream messages in the proxy. Maybe we should also unit-test the SSL upgrade that I recently implemented there.

### CODE REVIEW

File: components/electric/lib/electric/postgres/proxy/upstream_connection.ex [hunk 1]
```diff
@@ -206,4 +226,17 @@ defmodule Electric.Postgres.Proxy.UpstreamConnection do
...
+    :crypto.hash(:md5, iodata) |> bin_to_hex()
+  end
+
+  defp bin_to_hex(""), do: ""
+
+  defp bin_to_hex(<<hi::4, lo::4>> <> rest) do
+    <<hex_char(hi), hex_char(lo)>> <> bin_to_hex(rest)
+  end
+
+  defp hex_char(num) when num in 0..9, do: ?0 + num
```
Comment (icehaunter) [inline components/electric/lib/electric/postgres/proxy/upstream_connection.ex:240]: why not `Integer.to_string(num, 16)`? Speed hardly matters here for 2 chars?

Comment (icehaunter) [reply components/electric/lib/electric/postgres/proxy/upstream_connection.ex:240]: Actually, sorry, the entire `bin_to_hex` is replaceable with [`Base.encode16/2`](https://hexdocs.pm/elixir/Base.html#encode16/2)

**Review by icehaunter — APPROVED** [review]
LGTM with one nitpick, we should probs use a built-in functions

File: components/electric/lib/electric/postgres/proxy/upstream_connection.ex (hunk 1 above)
Comment (alco) [reply components/electric/lib/electric/postgres/proxy/upstream_connection.ex]: Perfect! Thanks!

**Pull request merged**